pymc_marketing/mmm/mmm.py

```diff
@@ -982,8 +1050,13 @@ def get_channel_contributions_forward_pass_grid(
             channel_data = (
                 delta * self.preprocessed_data["X"][self.channel_columns].to_numpy()
             )
+            time_index = (
+                np.arange(0, self.preprocessed_data["X"].shape[0])
+                if self.time_varying_media
+                else None
+            )
```

Comment:
Will we know if the dates are correct? Is this only used for the insampel data? What if the `channel_contributions` input are from a subset of the input. We will likely need to use the `infer_time_index` if this is used for new data. This a complaint of mine from the original implementation. If it is always used in the context of the insample, can we not just use the self._time_index variables create in `fit` call.

Reply:
This is a bit trickier. Should we